dashboard/app/models/plc/enrollment_unit_assignment.rb

```diff
@@ -68,6 +68,35 @@ def focus_area_positions
     plc_module_assignments.map{ |a| a.plc_learning_module.stage.position unless a.plc_learning_module.is_required? }.compact
   end
 
+  def summarize_progress
+    summary = []
+
+    categories_for_stage = plc_course_unit.script.stages.map(&:flex_category).uniq
+
+    if plc_course_unit.script.professional_learning_course?
+      Plc::LearningModule::MODULE_TYPES.keep_if { |type| categories_for_stage.include?(type)}.each do |flex_category|
```

Comment:
Good catch, @ashercodeorg. This looks unintentional. It's trying to select a subset of all module types (without mutating the constant of all MODULE_TYPES). 

@mehalshah It looks like `select` is the non-destructive version of `keep_if` that will return a copy of the array. See https://ruby-doc.org/core-2.2.0/Array.html#class-Array-label-Destructive+Selection